src/content/insertable-streams/webgpu/js/multi_video_main.js

```diff
@@ -0,0 +1,274 @@
+
+'use strict';
+
+const wgslShaders = {
+    vertex: `
+struct VertexInput {
+  [[location(0)]] position : vec3<f32>;
+  [[location(1)]] uv : vec2<f32>;
+};
+
+struct VertexOutput {
+  [[builtin(position)]] Position : vec4<f32>;
+  [[location(0)]] fragUV : vec2<f32>;
+};
+
+[[stage(vertex)]]
+fn main(input : VertexInput) -> VertexOutput {
+    var output : VertexOutput;
+    output.Position = vec4<f32>(input.position, 1.0);
+    output.fragUV = vec2<f32>(-0.5,-0.0) + input.uv;
+    return output;
+}
+`,
+
+    fragment: `
+[[binding(0), group(0)]] var mySampler: sampler;
+[[binding(1), group(0)]] var myTexture: texture_2d<f32>;
+
+[[stage(fragment)]]
+fn main([[location(0)]] fragUV : vec2<f32>) -> [[location(0)]] vec4<f32> {
+  return textureSample(myTexture, mySampler, fragUV);
+}
+`,
+};
+
+class WebGPUTransform {
+    constructor() {
+        this.canvas_ = null;
+        this.context_ = null;
+        this.device_ = null;
+        this.renderPipeline_ = null;
+        this.sampler_ = null;
+        this.videoTexture_ = null;
+        this.vertexBuffer_ = null;
+    }
+
+    async init() {
+        console.log('[WebGPUTransform] Initializing WebGPU.');
+
+        if (!this.canvas_) {
+            this.canvas_ = document.createElement('canvas');
+            document.getElementById('outputVideo').append(this.canvas_);
+            this.canvas_.width = 960;
+            this.canvas_.height = 540;
+        }
+
+        const canvas = this.canvas_;
+        const context = canvas.getContext('webgpu');
+        if (context === null || context === undefined) {
```

Comment:
nit: `if (!context)`

Same comment in several other places where you're checking if objects are null or undefined.